state-chain/custom-rpc/src/lib.rs

```diff
@@ -1085,6 +1129,25 @@ where
 		self.new_update_subscription(sink, move |api, hash| api.cf_pool_price(hash, from, to))
 	}
 
+	fn cf_subscribe_pool_price_v2(
+		&self,
+		sink: SubscriptionSink,
+		base_asset: RpcAsset,
+		quote_asset: RpcAsset,
+	) -> Result<(), SubscriptionEmptyError> {
+		let base = base_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
+		let quote = quote_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
+		self.new_update_subscription(sink, move |api, hash| {
```

Comment:
Also can we use `new_items_subscription`, I think we will remove `new_update_subscription` when we can in the future, as we have complaints that it is unhelpful when the subs don't return every block.

Reply:
Actually we'll need to use a version "new_items_subscription", but one that includes an initial item like "new_update_subscription" does.

Reply:
So either add a new "new_items_subscription_with_initial" function or maybe add a boolean parameter to "new_items_subscription" that allows you to control if there is an initial item.

Reply:
Oh I thought it was a bug to spam every block, but it looks like it is actually desired behaviuor